android/upload_symbols.gradle

```diff
@@ -0,0 +1,246 @@
+import java.util.zip.ZipOutputStream
+import java.util.zip.ZipEntry
+
+ext.boundary = 'abcd' + Long.toString(System.currentTimeMillis()) * 2 + 'dcba'
+ext.lineEnd = '\r\n'
+
+/**
+  * task to upload symbols to Instabug endpoint
+*/
+task upload_symbols_task {
+    doLast {
+        try {
+            String symbolsPath
+            String applicationToken
+            String apiKey
+            List config
+
+            // If obfuscate flag not provided, exit the task
+            if (!project.hasProperty('dart-obfuscation') || project['dart-obfuscation'] == 'false') {
+                return
+            }
+
+            config = get_config()
+            if (!config) {
+                return
+            }
+            (symbolsPath, applicationToken, apiKey) = config
+
+            // Prepare zip file with symbols
+            File zipFile = prepare_zip_file(symbolsPath)
+            if (!zipFile) {
+                return
+            }
+
+            upload_symbols(zipFile, applicationToken, apiKey)
+        } catch (Exception e) {
+            println("Instabug err: uploading symbols: ${e.message}")
+        }
+    }
+}
+/**
+    * get upload configs
+    * @return symbolsPath the path of the symbols directory
+    * @return applicationToken the application token
+    * @return apiKey the api key
+*/
+
+def get_config() {
```

Comment:
It's better if we split this function into two pure functions `getVariable(properties, name)` and `get symbolsPath()`, and use them in the `task` directly. This would be more concise than returning a `List` and depending that the caller uses them in the same order.